### PR TITLE
Feature diagrams should scroll automatically if a feature or constraint is moved to the border of the visible area & feature order drag&drop

### PR DESCRIPTION
**Pull request by Henningson**
enlarge the areas which are sensitive to DND mouse hover
allow autoscroll when the pointer is out of the viewport

### CODE REVIEW

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FullViewportAutoexposeHelper.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ * Autoexposer which allows the pointer to be outside of the viewport
+ * 
+ * Refactered version of VIewportAutoexposeHelper. Does not implement AutoexposeHelper by itself because ViewportHelper is not visible.
+ */
+public class FullViewportAutoexposeHelper extends ViewportAutoexposeHelper {
+
+	/**
+	 * Simple stop watch
+	 * 
+	 * TODO: Is there any equivalent class in the APIs?
```
Comment (Subaro) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FullViewportAutoexposeHelper.java]: If you only use the timer one time you can write the functionality directly into the code.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FullViewportAutoexposeHelper.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		}
+
+		Viewport port = findViewport(owner);
+		Point oldLoc = port.getViewLocation();
+		Point newLoc = getNewLocation(oldLoc, getInnerBounds().getPosition(point), getOffset());
+		port.setViewLocation(newLoc);
+		return true;
+	}
+
+	private boolean shouldScroll(Point point) {
```
Comment (Subaro) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FullViewportAutoexposeHelper.java]: Missing documentation

**Comment by Subaro** [comment]
> Also with the new root edit part, the auto layout of the legend is shifted into the constraints

# Current Develop

develop

# Your Changes

pull

**Comment by Subaro** [comment]
Okay, Thomas told me that I can´t drag the features/constraints out of the view and need to stop before the border. It works.

You should now fix the layout of the legend.

**Comment by 3dik** [comment]
This should fix the problems you have noted.

**Comment by 3dik** [comment]
Sorry for the extra commits. The following commits implement the "feature order DND":
* Change eclipse.List to eclipse.Table in class FeatureOrderEditor
* Enable DragAndDrop in the Table for FeatureOrderEditor
* Merge branch 'DragDropFeatureOrder' into autoscroll2

**Pull request merged**